### ISSUE TITLE
"root" path duplicated for new modules when changing "root" path of .angular-cli.json `ng generate`  to more then one subdirectory level

### ISSUE THREAD

**Issue by zladuric** [angular/angular-cli#8888]
### Versions
Windows 10, Node 8.9.2, npm 5.5.1, ng-cli 1.6.0:

```
[duricz@msgn07241 ~/temp/my-new-app]$ uname -a
MINGW64_NT-10.0 msgn07241 2.9.0(0.318/5/3) 2017-10-05 15:05 x86_64 Msys
[duricz@msgn07241 ~/temp/my-new-app]$ node -v
v8.9.2
[duricz@msgn07241 ~/temp/my-new-app]$ npm -v
5.5.1
[duricz@msgn07241 ~/temp/my-new-app]$ ng -v
_                      _                 ____ _     ___
/ \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
/ △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
/ ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
/_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
|___/

Angular CLI: 1.6.0
Node: 8.9.2
OS: win32 x64
Angular: 5.1.1
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.6.0
@angular-devkit/build-optimizer: 0.0.36
@angular-devkit/core: 0.0.22
@angular-devkit/schematics: 0.0.42
@ngtools/json-schema: 1.1.0
@ngtools/webpack: 1.9.0
@schematics/angular: 0.1.11
@schematics/schematics: 0.0.11
typescript: 2.4.2
webpack: 3.10.0

```

### Repro steps

If I change the `app.root` property to more than one level bellow "src" in `.angular-cli.json`, `ng generate` duplicates the source path when generating new components.

I mean, if "root" is "src", all is well. If root is "src/subdirectory", still all cool. If root is "src/sub/secondsubdirectory", the path of generated modules is duplicated.

Steps to reproduce:

* create a new app (`ng new my-new-app`)
* generate a new module to make sure module is done properly
* change the "root" property of .angular-cli.json to "src/main/frontend"
* generate a new module.

The path that is generated is duplicating the "root" part of the path.

### Observed behavior
```
duricz@msgn07241 ~/temp/my-new-app]$ grep root .angular-cli.json
      "root": "src",
[duricz@msgn07241 ~/temp/my-new-app]$ ng g m module
  create src/app/module/module.module.ts (190 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$ vi .angular-cli.json
[duricz@msgn07241 ~/temp/my-new-app]$ grep root .angular-cli.json
      "root": "src/main/frontend",
[duricz@msgn07241 ~/temp/my-new-app]$ ng g m another
  create src/main/frontend/src/main/frontend/app/another/another.module.ts (191 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$ ng g m another --flat
  create src/main/frontend/src/main/frontend/app/another.module.ts (191 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$
[duricz@msgn07241 ~/temp/my-new-app]$ ng g m nested/module
  create src/main/frontend/src/main/frontend/app/nested/module/module.module.ts (190 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$
```

### Desired behavior

I expect that my new module would be generated directly under `src/main/frontend/app`. Instead, my new module is made in `src/main/frontend/src/main/frontend/app`.

### Mention any other details that might be useful (optional)

This was mentioned in https://github.com/angular/angular-cli/issues/58, but closed.

The same happens if I try to generate service, component etc. The problem only appears for more than two levels of directory:

```
[duricz@msgn07241 ~/temp/my-new-app]$ vi .angular-cli.json
[duricz@msgn07241 ~/temp/my-new-app]$ ng g module mymo
  create srcss/app/mymo/mymo.module.ts (188 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$ vi .angular-cli.json
[duricz@msgn07241 ~/temp/my-new-app]$ ng g module mymo
  create src/single/app/mymo/mymo.module.ts (188 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$ vi .angular-cli.json
[duricz@msgn07241 ~/temp/my-new-app]$ ng g module mymo
  create src/single/twolevels/src/single/twolevels/app/mymo/mymo.module.ts (188 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$ vi .angular-cli.json
[duricz@msgn07241 ~/temp/my-new-app]$ ng g module mymo
  create src/single/twolevels/three/src/single/twolevels/three/app/mymo/mymo.module.ts (188 bytes)
[duricz@msgn07241 ~/temp/my-new-app]$
```

**Comment by ntrp** [comment]
Any news on this, it's kind of annoying to move all the generated code manually every time..

**Comment by zladuric** [comment]
Oh, I forgot to update this ticket. This started working for me, sometimes between 1.6.0 and 1.6.4, not sure which version fixed it. I will leave it open if you guys wanna make sure it works. @ntrp, did you try with the latest versions of ng-cli?

**Comment by ntrp** [comment]
I used 1.6.7 and it was happening.. for now I moved my sources close to the root to avoid the problem and to refactor the project.

**Comment by hansl** [comment]
Closing this as it should be addressed by the better workspace file in 6.0.
